pkg/gameservers/controller.go

```diff
@@ -639,6 +639,50 @@ func (c *Controller) addGameServerHealthCheck(gs *agonesv1.GameServer, pod *core
 	})
 }
 
+func (c *Controller) addSDKEnvironmentVarsToGameServer(gs *agonesv1.GameServer, pod *corev1.Pod) {
+	gs.ApplyToPodGameServerContainer(pod, func(c corev1.Container) corev1.Container {
+		sdkEnvVars := sdkEnvironmentVariables(gs)
+		if sdkEnvVars == nil {
+			// If a gameserver was created before 1.1 when we started defaulting the grpc and http ports,
+			// don't change the container spec.
+			return c
+		}
+
+		// Filter out environment variables that have reserved names.
+		// From https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating
+		env := c.Env[:0]
+		for _, e := range c.Env {
+			if !reservedEnvironmentVariableName(e.Name) {
+				env = append(env, e)
+			}
+		}
+		env = append(env, sdkEnvVars...)
+		c.Env = env
+		return c
+	})
+}
+
+func reservedEnvironmentVariableName(name string) bool {
+	return name == "AGONES_SDK_GRPC_PORT" || name == "AGONES_SDK_GRPC_PROXY_PORT"
+}
+
+func sdkEnvironmentVariables(gs *agonesv1.GameServer) []corev1.EnvVar {
+	var env []corev1.EnvVar
+	if gs.Spec.SdkServer.GRPCPort != 0 {
+		env = append(env, corev1.EnvVar{
+			Name:  "AGONES_SDK_GRPC_PORT",
+			Value: strconv.Itoa(int(gs.Spec.SdkServer.GRPCPort)),
+		})
+	}
+	if gs.Spec.SdkServer.HTTPPort != 0 {
+		env = append(env, corev1.EnvVar{
+			Name:  "AGONES_SDK_GRPC_PROXY_PORT",
```

Comment:
```suggestion
			Name:  "AGONES_SDK_HTTP_PORT",
```
?

Or REST_PORT ?

Not a strong opinion, just a thought.

Reply:
Changed the variable and made it into a constant.